src/agent/application.cpp

```diff
@@ -255,4 +199,89 @@ void Application::HandleSignal(int aSignal)
     signal(aSignal, SIG_DFL);
 }
 
+void Application::ConstructRcpMode(const std::string &aRestListenAddress, int aRestListenPort)
+{
+    otbr::Ncp::RcpHost &rcpHost = static_cast<otbr::Ncp::RcpHost &>(*mHost);
```

Comment:
Ideally we should avoid down-casting. Is there a plan to components accepting ThreadController instead of RcpHost?

Reply:
Yes, ideally we hope all the components work with the interface `ThreadController` instead of the implementation class. But at this stage, there are too many dependencies on `RcpHost` so we still let them depend on `RcpHost`. Later we will gradually change the dependence of `RcpHost` to `ThreadController`.